service/src/main/java/com/ericsson/eiffel/remrem/generate/controller/RemremGenerateController.java

```diff
@@ -78,84 +77,221 @@ public class RemremGenerateController {
-                response = msgService.generateMsg(msgType, bodyJson, isLenientEnabled(okToLeaveOutInvalidOptionalFields));
-                JsonElement parsedResponse = parser.parse(response);
-                if(lookupLimit <= 0) {
-                    return new ResponseEntity<>("LookupLimit must be greater than or equals to 1", HttpStatus.BAD_REQUEST);
+            if (lookupLimit <= 0) {
+                return new ResponseEntity<>("Parameter 'lookupLimit' must be > 0", HttpStatus.BAD_REQUEST);
+            }
+            if (inputData == null) {
+                return createResponseEntity(HttpStatus.BAD_REQUEST, "Parameter 'inputData' must not be null",
+                        JSON_ERROR_STATUS);
+            }
+
+            if (inputData.isJsonArray()) {
+                JsonArray inputEventJsonArray = inputData.getAsJsonArray();
+                for (JsonElement element : inputEventJsonArray) {
+                    JsonObject generatedEvent = (processEvent(msgProtocol, msgType,
+                            failIfMultipleFound, failIfNoneFound, lookupInExternalERs, lookupLimit,
+                            okToLeaveOutInvalidOptionalFields, element.getAsJsonObject()));
+                    generatedEventResults.add(generatedEvent);
+                }
+                boolean success = true;
+                for (JsonElement result : generatedEventResults) {
+                    JsonObject jsonObject = result.getAsJsonObject();
+                    success &= jsonObject.has(META);
                 }
-                if (!parsedResponse.getAsJsonObject().has(RemremGenerateServiceConstants.JSON_ERROR_MESSAGE_FIELD)) {
-                    return new ResponseEntity<>(parsedResponse, HttpStatus.OK);
+                return new ResponseEntity<>(generatedEventResults, success ? HttpStatus.OK : HttpStatus.BAD_REQUEST);
+
+            } else if (inputData.isJsonObject()) {
+                JsonObject inputJsonObject = inputData.getAsJsonObject();
+                JsonObject processedJson = processEvent(msgProtocol, msgType, failIfMultipleFound, failIfNoneFound,
+                        lookupInExternalERs, lookupLimit, okToLeaveOutInvalidOptionalFields, inputJsonObject);
+                HttpStatus status;
+                String statusValue = null;
+                if (processedJson.has(META)) {
+                    status = HttpStatus.OK;
+                    return new ResponseEntity<>(processedJson, status);
+                } else if (processedJson.has(JSON_STATUS_CODE)) {
+                    statusValue = processedJson.get(JSON_STATUS_CODE).toString();
+                    try {
+                        status = HttpStatus.resolve(Integer.parseInt(statusValue));
+                        return new ResponseEntity<>(processedJson, status);
+                    } catch (NumberFormatException e) {
+                        log.error("Invalid status value: '" + statusValue + "' of response " + processedJson);
```

Comment:
Why to compose the same string twice? Store it in a variable and reuse it. The same 195 and 196.

Reply:
Can do like this?

 **else {
                    String errorMessage = "There is no status value in the response " + processedJson;
                    log.error(errorMessage);
                    return createResponseEntity(HttpStatus.INTERNAL_SERVER_ERROR, errorMessage, JSON_ERROR_STATUS);
                }**

Reply:
Yes, that's perfect solution.